app/Models/EntryDAO.php

```diff
@@ -1001,68 +909,67 @@ public function updateLastSeen($id_feed, $guids, $mtime = 0) {
+		$stm = $this->pdo->query($sql);
+		if ($stm === false) {
 			return false;
 		}
-		$stm->execute();
 		$res = $stm->fetchAll(PDO::FETCH_COLUMN, 0);
 		rsort($res);
 		$all = empty($res[0]) ? 0 : $res[0];
 		$unread = empty($res[1]) ? 0 : $res[1];
 		return array('all' => $all, 'unread' => $unread, 'read' => $all - $unread);
 	}
+
 	public function count($minPriority = null) {
-		$sql = 'SELECT COUNT(e.id) AS count FROM `' . $this->prefix . 'entry` e';
+		$sql = 'SELECT COUNT(e.id) AS count FROM `_entry` e';
 		if ($minPriority !== null) {
-			$sql .= ' INNER JOIN `' . $this->prefix . 'feed` f ON e.id_feed=f.id';
+			$sql .= ' INNER JOIN `_feed` f ON e.id_feed=f.id';
 			$sql .= ' WHERE f.priority > ' . intval($minPriority);
 		}
-		$stm = $this->bd->prepare($sql);
-		$stm->execute();
+		$stm = $this->pdo->query($sql);
 		$res = $stm->fetchAll(PDO::FETCH_COLUMN, 0);
 		return isset($res[0]) ? $res[0] : 0;
 	}
+
 	public function countNotRead($minPriority = null) {
-		$sql = 'SELECT COUNT(e.id) AS count FROM `' . $this->prefix . 'entry` e';
+		$sql = 'SELECT COUNT(e.id) AS count FROM `_entry` e';
 		if ($minPriority !== null) {
-			$sql .= ' INNER JOIN `' . $this->prefix . 'feed` f ON e.id_feed=f.id';
+			$sql .= ' INNER JOIN `_feed` f ON e.id_feed=f.id';
 		}
 		$sql .= ' WHERE e.is_read=0';
 		if ($minPriority !== null) {
 			$sql .= ' AND f.priority > ' . intval($minPriority);
 		}
-		$stm = $this->bd->prepare($sql);
-		$stm->execute();
+		$stm = $this->pdo->query($sql);
 		$res = $stm->fetchAll(PDO::FETCH_COLUMN, 0);
 		return $res[0];
 	}
 
 	public function countUnreadReadFavorites() {
-		$sql = <<<SQL
-  SELECT c
-    FROM (
-         SELECT COUNT(e1.id) AS c
-              , 1 AS o
-           FROM `{$this->prefix}entry` AS e1
-           JOIN `{$this->prefix}feed` AS f1 ON e1.id_feed = f1.id
-          WHERE e1.is_favorite = 1
-            AND f1.priority >= :priority_normal
-         UNION
-         SELECT COUNT(e2.id) AS c
-              , 2 AS o
-           FROM `{$this->prefix}entry` AS e2
-           JOIN `{$this->prefix}feed` AS f2 ON e2.id_feed = f2.id
-          WHERE e2.is_favorite = 1
-            AND e2.is_read = 0
-            AND f2.priority >= :priority_normal
-         ) u
+		$sql = <<<'SQL'
+SELECT c FROM (
+	SELECT COUNT(e1.id) AS c, 1 AS o
+		 FROM `_entry` AS e1
+		 JOIN `_feed` AS f1 ON e1.id_feed = f1.id
+		WHERE e1.is_favorite = 1
+		  AND f1.priority >= :priority_normal1
+	UNION
+	SELECT COUNT(e2.id) AS c, 2 AS o
+		 FROM `_entry` AS e2
+		 JOIN `_feed` AS f2 ON e2.id_feed = f2.id
+		WHERE e2.is_favorite = 1
+		  AND e2.is_read = 0
+		  AND f2.priority >= :priority_normal2
+	) u
 ORDER BY o
 SQL;
-		$stm = $this->bd->prepare($sql);
-		$stm->execute(array(':priority_normal' => FreshRSS_Feed::PRIORITY_NORMAL));
+		$stm = $this->pdo->prepare($sql);
+		//Binding a value more than once is not standard and does not work with native prepared statements (e.g. MySQL) https://bugs.php.net/bug.php?id=40417
```

Comment:
Oo this is very not intuitive! Does it mean previous code wasn't working correctly?

Reply:
It means the previous code was working just by chance, because the current PDO default behaviour for MySQL is to use emulated prepared statements instead of native, and because the PHP implementation accidentality allowed that. It has never been standard